code/datums/datum.dm

```diff
@@ -58,6 +58,15 @@
 
 	return QDEL_HINT_QUEUE
 
+/datum/proc/CalcMemory()
+	. = 24
+	var/list/builtins = list(NAMEOF(src, type) = TRUE, NAMEOF(src, vars) = TRUE, NAMEOF(src, tag) = TRUE, NAMEOF(src, parent_type) = TRUE)

```

Comment:
You should include ALL builtins, not just datum builtins, and remove the NAMEOF bs, there is no way to do this properly that doesn't involve overriding this for every type and it would be easier and better to just forgot type safety for a var blacklist since a var not existing is not an error state strictly speaking.

Also it might be faster to do `vars-builtin`, allowing you to forgo the associated part. Look at whatever I did in the findReferences TESTING proc in ssgarbage for the var blacklist there, as that was what profiled fastest.